db/table_cache.cc

```diff
@@ -404,7 +404,91 @@ Status TableCache::MultiGet(const ReadOptions& options,
   Status s;
   TableReader* t = fd.table_reader;
   Cache::Handle* handle = nullptr;
-  if (s.ok()) {
+  MultiGetRange table_range(*mget_range, mget_range->begin(), mget_range->end());
+#ifndef ROCKSDB_LITE
+  autovector<std::string, MultiGetContext::MAX_BATCH_SIZE> row_cache_entries;
```

Comment:
We could consider factoring out this logic into separate helper method(s) to eliminate the code duplication between Get() and MultiGet().

Reply:
Let me see how this can be refactored. Need to be careful not to add overhead to Get() due to bookkeeping overhead of working with a batch.